File: Assignment12/Q4.c
/////////////////////////////////////////////////////////////////////////////////
//
//      Include required header files
//
/////////////////////////////////////////////////////////////////////////////////
#include<stdio.h>

/////////////////////////////////////////////////////////////////////////////////
//
//      Function name:  MultDigits
//      Input:          integer
//      Output:         integer
//      Description:    Accept number from user and return multiplication
//                      of all digits
//      Date:           01/06/2026
//      Author:         Pratik Subhash Mahale
//
/////////////////////////////////////////////////////////////////////////////////
int MultDigits(
                    int iNo            // Input number
              )
{
    int iDigit = 0;
    int iMult = 1;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    while(iNo > 0)
    {
        iDigit = iNo % 10;

        if(iDigit != 0)
        {
            iMult = iMult * iDigit;
        }

        iNo = iNo / 10;
    }

    return iMult;
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Application to calculate multiplication of digits
//
/////////////////////////////////////////////////////////////////////////////////
int main()
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter number\n");

    scanf("%d", &iValue);

    iRet = MultDigits(iValue);

    printf("%d", iRet);

    return 0;
}

/////////////////////////////////////////////////////////////////////////////////
//
//      Input:      2395
//      Output:     270
//
//      Input:      1018
//      Output:     8
//
//      Input:      9440
//      Output:     144
//
//      Input:      922432
//      Output:     864
//
/////////////////////////////////////////////////////////////////////////////////

/////////////////////////////////////////////////////////////////////////////////
//
//      Time Complexity: O(n)
//      Space Complexity: O(1)
//
//      where n = number of digits in the input number
//
/////////////////////////////////////////////////////////////////////////////////

File: Assignment18/Q4.java

///////////////////////////////////////////////////////////////////
//
//      File Name : Q4.java
//      Project   : Logic Building Assignment 18
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Class Name : EvenOddDigitLogic
//      Method Name: sumEvenOddDigits
//      Input      : Integer
//      Output     : Void
//      Description: Find the sum of even digits and
//                   sum of odd digits separately
//      Date       : 08/06/2026
//      Author     : Pratik Subhash Mahale
//
///////////////////////////////////////////////////////////////////

class EvenOddDigitLogic
{
    void sumEvenOddDigits(int num)
    {
        int digit = 0;      // Stores extracted digit
        int evenSum = 0;    // Stores sum of even digits
        int oddSum = 0;     // Stores sum of odd digits

        // If number is negative convert it to positive
        if(num < 0)
        {
            num = -num;
        }

        // Extract digits and calculate sums
        while(num != 0)
        {
            digit = num % 10;

            if(digit % 2 == 0)
            {
                evenSum = evenSum + digit;
            }
            else
            {
                oddSum = oddSum + digit;
            }

            num = num / 10;
        }

        // Display results
        System.out.println("Sum of Even Digits : " + evenSum);
        System.out.println("Sum of Odd Digits  : " + oddSum);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Main Class
//
///////////////////////////////////////////////////////////////////

public class Q4
{
    public static void main(String args[])
    {
        // Create object of EvenOddDigitLogic class
        EvenOddDigitLogic obj = new EvenOddDigitLogic();

        // Function call
        obj.sumEvenOddDigits(123456);
    }
}

///////////////////////////////////////////////////////////////////
//
//      Input  : 123456
//
//      Output :
//      Sum of Even Digits : 12
//      Sum of Odd Digits  : 9
//
///////////////////////////////////////////////////////////////////

///////////////////////////////////////////////////////////////////
//
//      Time Complexity : O(d)
//
//      Space Complexity : O(1)
//
//      Where d = Number of digits
//
///////////////////////////////////////////////////////////////////
